frontend/src/app/shared/services/error-handler.service.ts

```diff
@@ -0,0 +1,29 @@
+import { ErrorHandler, Injectable } from '@angular/core';
+import { AlertService } from './alert.service';
+import { AlertType } from '../components/alert/alert.component';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Injectable({
+    providedIn: 'root'   
+})
+export class ErrorHandlerService implements ErrorHandler {
```

Comment:
This is great! But maybe specify that this service is specific for HTTP errors, or make it more generic?

Reply:
Yes, good idea! I'll add a class method for generic error messages so we can use it for other errors/notifications as well.